### PR TITLE
Issue-10: Add Buddy PR Checks

### PR DESCRIPTION
**Pull request by kevinfodness**
### Description

Add a configuration for PR checks using Buddy, which performs the following checks against an assumed plugin and theme:

- gitignored files check (ensure files that should be ignored are not committed)
- git conflict markers check (ensure merge conflicts aren't committed)
- phpcs
- phpstan
- phpunit
- npm build
- lint (includes ESLint, StyleLint, TypeScript typechecking)
- JS tests using jest

### Use Case

When a user uses this repository to build a project, they should be able to automate PR checks using Buddy.

### CODE REVIEW

**Review by efuller — APPROVED** [review]
⚡ Learned some things about Buddy!

File: .npmrc [hunk 1]
```diff
@@ -1 +1 @@
-auto-install-peers = true
+legacy-peer-deps=true
```
Comment (kaitlinbolling) [inline .npmrc:1]: What's requiring this? I think ideally we wouldn't need it.

**Review by kaitlinbolling — APPROVED** [review]
👍  Looks good! Just one question.

File: .npmrc (hunk 1 above)
Comment (kevinfodness) [reply .npmrc:1]: The WordPress dependencies. They're working on fixing it but that fix isn't out yet. It's this problem: https://github.com/WordPress/gutenberg/issues/48009 They're working on replacing `reakit` with `ariakit` but that's not complete yet. When it is, we can stop using legacy peer deps.

Comment (kevinfodness) [reply .npmrc:1]: Actually, apparently `reakit` is now fully removed, but we need to make sure that the versions of the dependencies we are loading for the version of WP that we're using are the ones with reakit removed. Hopefully that'll be true when WP 6.5 releases but we will need to confirm.

**Pull request merged**